### PR TITLE
Fix ignore_index in jaccard index

### LINKED ISSUE #1385: MulticlassJaccardIndex: wrong result with ignore_index
## 🐛 Bug

MulticlassJaccardIndex seems to ignore `ignore_index` in reduction.

### To Reproduce

Your example from docs:
```python
import torch
from torchmetrics.classification import MulticlassJaccardIndex

target = torch.tensor([2, 1, 0, 0])
preds = torch.tensor([2, 1, 0, 1])
metric = MulticlassJaccardIndex(num_classes=3)
metric(preds, target)
print(metric.confmat)
print(metric.compute())
```
The output is fine:
```text
tensor([[1, 1, 0],
        [0, 1, 0],
        [0, 0, 1]])
tensor(0.6667)
```

However, ignoring classes (e.g. void) seems to be broken:
```python
import torch
from torchmetrics.classification import MulticlassJaccardIndex

target = torch.tensor([2, 1, 0, 0])
preds = torch.tensor([2, 1, 0, 1])
metric = MulticlassJaccardIndex(num_classes=3, ignore_index=0)
metric(preds, target)
print(metric.confmat)
print(metric.compute())
```

```text
tensor([[0, 0, 0],
        [0, 1, 0],
        [0, 0, 1]])
tensor(0.6667)
```

### Expected behavior

Output should be 1.0 as class 0 is ignored.
Looks like predictions with targets of class 0 are correctly masked but the empty class 0 is still part of the reduction.

### Environment

- TorchMetrics version 0.10.2 / 0.11.0
- Python 3.8.8, PyTorch 1.13.0

### Additional context

Note that the old `JaccardIndex` in 0.10.2 was working as expected.

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #1385 

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@SkafteNicki is this ready for review? :rabbit:

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
